File: snippets/fsharp/VS_Snippets_CLR_System/system.int64.parse/fs/Parse1.fs
module Parse1

// <Snippet1>
open System

let convert value =
    try
        let number = Int64.Parse value
        printfn $"Converted '{value}' to {number}."
    with
    | :? FormatException ->
        printfn $"Unable to convert '{value}'."
    | :? OverflowException ->
        printfn $"'{value}' is out of range."

convert "  179042  "
convert " -2041326 "
convert " +8091522 "
convert "   1064.0   "
convert "  178.3"
convert String.Empty

decimal Int64.MaxValue + 1M
|> string
|> convert

// This example displays the following output to the console:
//       Converted '  179042  ' to 179042.
//       Converted ' -2041326 ' to -2041326.
//       Converted ' +8091522 ' to 8091522.
//       Unable to convert '   1064.0   '.
//       Unable to convert '  178.3'.
//       Unable to convert ''.
//       '92233720368547758071' is out of range.
// </Snippet1>

File: snippets/csharp/System/Int64/Parse/Parse1.cs
// <Snippet1>
using System;

public class Int64ParseExample1
{
    public static void Run()
    {
        Convert("  179042  ");
        Convert(" -2041326 ");
        Convert(" +8091522 ");
        Convert("   1064.0   ");
        Convert("  178.3");
        Convert(string.Empty);
        Convert(((decimal)long.MaxValue) + 1.ToString());
    }

    private static void Convert(string value)
    {
        try
        {
            long number = long.Parse(value);
            Console.WriteLine($"Converted '{value}' to {number}.");
        }
        catch (FormatException)
        {
            Console.WriteLine($"Unable to convert '{value}'.");
        }
        catch (OverflowException)
        {
            Console.WriteLine($"'{value}' is out of range.");
        }
    }
}
// This example displays the following output to the console:
//       Converted '  179042  ' to 179042.
//       Converted ' -2041326 ' to -2041326.
//       Converted ' +8091522 ' to 8091522.
//       Unable to convert '   1064.0   '.
//       Unable to convert '  178.3'.
//       Unable to convert ''.
//       '92233720368547758071' is out of range.
// </Snippet1>
